src/Access/SettingsConstraints.h

```diff
@@ -87,29 +82,60 @@ class SettingsConstraints
     friend bool operator !=(const SettingsConstraints & left, const SettingsConstraints & right) { return !(left == right); }
 
 private:
-    struct Constraint
+    enum ReactionOnViolation
+    {
+        THROW_ON_VIOLATION,
+        CLAMP_ON_VIOLATION,
+    };
+
+    struct Range
     {
-        std::shared_ptr<const String> setting_name;
-        bool read_only = false;
+        SettingConstraintType type = SettingConstraintType::NONE;
         Field min_value;
         Field max_value;
 
-        bool operator ==(const Constraint & other) const;
-        bool operator !=(const Constraint & other) const { return !(*this == other); }
+        String explain;
+        int code;
+
+        bool operator ==(const Range & other) const;
+        bool operator !=(const Range & other) const { return !(*this == other); }
+
+        bool check(SettingChange & change, const Field & new_value, ReactionOnViolation reaction) const;
+
+        static const Range & allowed()
+        {
+            static Range res;
+            return res;
+        }
+
+        static Range forbidden(const String & explain, int code)
+        {
+            return Range{.type = SettingConstraintType::CONST, .explain = explain, .code = code};
```

Comment:
It seems you use the structure `Range` for two independent thing: 1) information about a constraint and a value in the `constraints` map; and 2) as a part of constraint-checking algorithm and the return value of `SettingsConstraints::getRange`. It seems it would be better to separate this structure into two